streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/AssignmentInfo.java

```diff
@@ -196,13 +219,18 @@ private void writeTopicPartitions(final DataOutputStream out,
         }
     }
 
-    private void encodeVersionThree(final DataOutputStream out) throws IOException {
-        out.writeInt(3);
+    private void encodeVersionThree(int version, final DataOutputStream out) throws IOException {
```

Comment:
Ditto, I'd suggest just duplicating the code since we may add more logic for version 4 anyways.

Reply:
Addition to version 4 can be added at the end of `encodeVersionFour`

Reply:
Adding a parameter `version` for `encodeVersionThree` is very confusing to other readers. I'd suggest completely duplicate the code in `encodeVersionFour` and remove this parameter in `encodeVersionThree`.